### PR TITLE
docs(adr): correct ADR-057 eval-enforcement torn state (Deliverable B, Refs #3185)

### LINKED ISSUE #3185: feat(validation): enforce ADR-057 behavioral-eval evidence in pre_pr.py and CI (replaces inert prompt_eval_gate)
ADR-057 mandates behavioral eval evidence for prompt, agent, and skill changes, but its only automated enforcement point, the PreToolUse hook `.claude/hooks/PreToolUse/invoke_prompt_eval_gate.py`, has never blocked a single commit: it emits `{"decision":"deny"}` with exit 0, a payload the Claude Code harness ignores (the exact bug class documented and fixed in the sibling `invoke_security_commit_gate.py`, Issue #2521, but never fixed here). Since 2026-06-01, 108 commits touched ADR-057-scoped prompt files while exactly 1 commit added eval evidence to `.agents/eval-results/`. This issue makes the executor decide, with evidence, between two deliverables: (A) implement the gate once, deterministically, in `scripts/validation/pre_pr.py` plus a CI leg in `.github/workflows/ai-pr-quality-gate.yml`, or (B) amend ADR-057 to demote automated enforcement to the existing advisory reminder, deleting the false enforcement claims. Either way, the false "Claude hook provides blocking enforcement" claims in ADR-057 and `.githooks/pre-commit` get corrected.

## Problem

ADR-057 (`.agents/architecture/ADR-057-prompt-behavioral-evaluation.md`) declares in its Confirmation section (line 222):

> | Prompt/skill/agent changes require eval evidence before commit | invoke_prompt_eval_gate.py | Claude Code PreToolUse hook, blocks `git commit` |

That claim is false. The hook cannot block. Its docstring admits the shape (`.claude/hooks/PreToolUse/invoke_prompt_eval_gate.py:15-16`):

```text
Exit Codes (Claude Hook Semantics):
    0 = Always (uses JSON decision payload for deny/allow semantics)
```

and its deny path (lines 246-261) prints a top-level `{"decision": "deny", ...}` payload then `return 0`. The recognized Claude Code PreToolUse block contract is a top-level `{"decision": "block"}` paired with exit 2. The repo already documents this in the fixed sibling, quoted verbatim per `.claude/rules/canonical-source-mirror.md` from `.claude/hooks/PreToolUse/invoke_security_commit_gate.py:167-171`:

```text
    Claude Code recognizes a top-level ``{"decision": "block"}`` payload, paired
    with exit code 2, to deny a tool call. The previous ``{"decision": "deny"}``
    with exit 0 was not a recognized top-level value, so the gate's output was
    ignored and the security-sensitive commit proceeded (Issue #2521). The value
    ``deny`` is only valid under ``hookSpecificOutput.permissionDecision``.
```

`invoke_prompt_eval_gate.py` shipped with the same bug and was never fixed. Every commit in two harnesses (Claude Code and the generated Copilot CLI shim) pays a Python spawn plus a `git diff --cached` for a gate that cannot fire. Meanwhile the enforcement the ADR promises does not exist anywhere deterministic: `scripts/validation/pre_pr.py` (the mandated pre-PR gate per AGENTS.md) has no eval check, and `.github/workflows/ai-pr-quality-gate.yml` has no eval job. The hook itself is deleted by #3184; this issue owns the question of whether the ADR-057 requirement survives, and if so, where it is enforced.

Even if the payload were fixed in place, the hook's evidence heuristics are too weak to enforce anything (lines 139-178): any `.agents/eval-results/<today>-*.json` file unlocks every staged file regardless of which prompt it evaluated, any staged `*.json` with "eval" in the name passes, and any `scripts/eval/*.json` modified within 24 hours passes. Enforcement, if kept, needs a real change-to-evidence match.

## Evidence (verified 2026-07-17)

- Hook exists and is inert. `.claude/hooks/PreToolUse/invoke_prompt_eval_gate.py` (273 lines). Deny path at lines 246-261 builds `output = {"decision": "deny", "reason": ...}` then `print(json.dumps(output, ...))` and `return 0`. No exit 2 anywhere on the deny path.
- Registered in BOTH registration files, on the `git commit` matcher group:
  - `.claude/settings.json:78`: `"command": "python3 -u .claude/hooks/PreToolUse/invoke_prompt_eval_gate.py"` (with `"timeout": 10` at line 79, statusMessage at line 80).
  - `.claude/hooks/hooks.json:79`: `"command": "python3 -u \"${CLAUDE_PLUGIN_ROOT}/hooks/PreToolUse/invoke_prompt_eval_gate.py\""` (timeout at line 80, statusMessage at line 81).
- Generated Copilot shim exists: `src/copilot-cli/hooks/PreToolUse/invoke_prompt_eval_gate__Bash_gG_iI_tT_cC_oO_mM_mM_iI_tT_gG_iI_tT_cC_iI_g_ec29be.py`, listed in `src/copilot-cli/hooks/PreToolUse/_manifest.json` lines 14 and 43. Equally inert.
- Test references: `tests/test_hook_plugin_guards.py:93` lists the hook in `_COMMIT_GATE_SCRIPTS` (used by parametrized tests at lines 163 and 190). No dedicated behavior test file exists for this hook (`git grep -l invoke_prompt_eval_gate tests/` returns only `tests/test_hook_plugin_guards.py`).
- Enforcement gap is measurable. Commands run from repo root on 2026-07-17:
  - `git log --oneline --since=2026-06-01 -- '.claude/skills/*/SKILL.md' '.claude/commands/' '.claude/agents/' | wc -l` returned `108`.
  - `git log --oneline --since=2026-06-01 -- '.agents/eval-results/' | wc -l` returned `1`.
  - `.agents/eval-results/` holds exactly 4 files: `2026-04-19-issue-1700-orchestrator-checklist.json`, `2026-05-24-structural-only-skill-fixes.json`, `2026-05-31-issue-2001-spec-generator-skill.json`, `2026-07-08-ship-mode-detection.json`.
- ADR-057 contract, quoted verbatim per `.claude/rules/canonical-source-mirror.md`. Scope table (`.agents/architecture/ADR-057-prompt-behavioral-evaluation.md:57-64`):

  ```text
  | Category | Path Patterns |
  |----------|---------------|
  | Commands | `.claude/commands/*.md` |
  | Quality gate prompts | `.github/prompts/*.md` |
  | Security prompts | `.agents/security/prompts/*.md` |
  | Agent definitions (Claude Code) | `.claude/agents/*.md` |
  | Agent definitions (published) | `src/claude/*.md`, `src/copilot-cli/*.md`, `src/vs-code-agents/*.md` |
  | Skill definitions | `.claude/skills/*/SKILL.md` |
  ```

  Line 66 (verbatim, the backticks are in the source): ``Excluded from behavioral eval: `CLAUDE.md`, `README.md`, `INDEX.md`, `AGENTS.md`, and `.template.` files.`` Evidence storage (lines 147-150): security benchmarks in `.agents/security/benchmarks/`, other prompt evals in `tests/evals/`. Enforcement Path (lines 243-247) says `Current: Claude Code hook blocks commits` (false) and `Future: CI automation of eval runs. Deferred until eval runner stabilizes and cost model is validated.`
- The hook's compiled pattern list, the operational refinement of the ADR table (narrowed in #2052), quoted verbatim from `.claude/hooks/PreToolUse/invoke_prompt_eval_gate.py:63-99` (comment block elided at the `#` lines for brevity; copy from the file or from git history at the SHA current when you start if #3184 has already landed):

  ```python
  _PROMPT_PATTERNS = [
      re.compile(r"^\.claude/commands/.*\.md$"),
      re.compile(r"^\.github/prompts/.*\.md$"),
      re.compile(r"^\.agents/security/prompts/.*\.md$"),
      re.compile(r"^src/copilot-cli/docs/.*\.md$"),
  ]

  _AGENT_PATTERNS = [
      re.compile(r"^\.claude/agents/(?!CLAUDE\.md|README\.md|INDEX\.md).*\.md$"),
      re.compile(r"^src/claude/(?!CLAUDE\.md|README\.md|AGENTS\.md)(?!.*\.template\.).*\.md$"),
      re.compile(r"^src/copilot-cli/agents/.*\.agent\.md$"),
      re.compile(r"^src/copilot-cli/docs/.*\.md$"),
      re.compile(r"^src/vs-code-agents/.*\.md$"),
  ]

  _SKILL_PATTERNS = [
      re.compile(r"^\.claude/skills/.*/SKILL\.md$"),
  ]

  _SECURITY_CRITICAL_PATTERNS = [
      re.compile(r"security", re.IGNORECASE),
      re.compile(r"pr-quality-gate-security"),
  ]
  ```

- `.githooks/pre-commit` lines 2043-2087 already carry a NON-blocking ADR-057 reminder for the same three path classes, with this false comment at lines 2047-2049: `The Claude Code hook (invoke_prompt_eval_gate.py) provides the blocking enforcement for AI agents.` and at line 2086: `# Non-blocking: reminder only, Claude hook provides enforcement`.
- `scripts/validation/pre_pr.py` is a thin runner (decomposed per issue #2223): validators live in sibling `checks_*` modules, each wired via `run_validation(...)` calls inside `main()` (first wiring at lines 272-276). `pre_pr.py` runs locally only; the sole workflow mention of it is a comment in `.github/workflows/validate-generated-agents.yml:191`. CI enforcement therefore requires its own leg.
- `.github/workflows/ai-pr-quality-gate.yml` (735 lines) is the anchor for a CI leg: jobs are thin and delegate to Python per ADR-006 (for example `python3 scripts/quality_gate/run_pytest.py` at line 235, `python3 build/scripts/run_drift_check_ci.py` at line 195).
- Eval runners exist and are runnable today: `scripts/eval/eval-prompt-change.py` and `scripts/eval/eval-suite.py` (both present in `scripts/eval/`).

## Requirements (EARS)

- REQ-1: WHEN the executor starts this issue, THE SYSTEM SHALL read `.agents/architecture/ADR-057-prompt-behavioral-evaluation.md` in full and rerun the two `git log` evidence commands above, then choose exactly one deliverable, A (deterministic gate) or B (ADR amendment demoting enforcement), and record the choice with rationale in the PR description, SO THAT enforcement matches actual practice instead of shipping a gate nobody honors or an ADR nobody enforces.
- REQ-2: IF Deliverable A is chosen, WHEN `python3 scripts/validation/pre_pr.py` runs on a branch whose diff against the base ref (resolve via `_resolve_branch_base_ref` in `scripts/validation/checks_common.py`) contains files matching the pattern lists quoted above, THE SYSTEM SHALL fail the run (contributing to exit 1) unless matching eval evidence exists, SO THAT ADR-057-scoped changes cannot reach a PR without behavioral eval evidence.
- REQ-3: IF Deliverable A is chosen, THE SYSTEM SHALL count as evidence only a `.agents/eval-results/*.json` file that is itself part of the same branch diff (added or modified since the merge base), SO THAT the inert hook's gameable heuristics (any today-dated file, any staged JSON containing "eval", any fresh `scripts/eval/*.json`) are not reproduced.
- REQ-4: IF Deliverable A is chosen, THE SYSTEM SHALL use path patterns character-for-character identical to the hook's `_PROMPT_PATTERNS`, `_AGENT_PATTERNS`, and `_SKILL_PATTERNS` lists quoted above, cite `invoke_prompt_eval_gate.py` (path plus, once deleted, the deletion SHA) in the module docstring, and document any intentional divergence per `.claude/rules/canonical-source-mirror.md`, SO THAT the enforced scope is the reviewed ADR-057 scope, not a reinvention.
- REQ-5: IF Deliverable A is chosen, WHEN the environment variable `SKIP_PROMPT_EVAL=1` is set, THE SYSTEM SHALL pass the check while printing a loud warning naming the bypass, SO THAT the documented ADR-057 escape hatch survives with an audit trail.
- REQ-6: IF Deliverable A is chosen, WHEN a pull request changes ADR-057-scoped paths, THE SYSTEM SHALL run the same check in `.github/workflows/ai-pr-quality-gate.yml` as a single-command step delegating to the Python module (ADR-006: no logic in YAML), SO THAT enforcement holds even when an agent or human skips the local `pre_pr.py` run.
- REQ-7: WHEN this issue lands (either deliverable), THE SYSTEM SHALL replace the false claims in `.githooks/pre-commit` (lines 2047-2049 and 2086) with text naming the actual enforcement point (Deliverable A: `pre_pr.py` plus CI; Deliverable B: advisory reminder only), SO THAT the next reader is not told a dead hook protects them.
- REQ-8: WHEN this issue lands (either deliverable), THE SYSTEM SHALL amend ADR-057 (Confirmation table line 222 and Enforcement Path lines 243-247) to name the real enforcement mechanism or, for Deliverable B, to record with the 108-vs-1 data that automated pre-commit enforcement is retired and evidence review moves to code review, SO THAT the ADR stops asserting an enforcement that does not exist. The edit fires the adr-review skill per AGENTS.md.
- REQ-9: IF Deliverable A is chosen, THE SYSTEM SHALL ship pytest coverage in `tests/validation/` with positive (scoped change plus branch-diff evidence passes), negative (scoped change without evidence fails; bypass env var passes with warning), and edge cases (excluded files like `.template.` and `README.md` do not trigger; evidence file outside the branch diff does not count), mocking git subprocess boundaries, SO THAT the gate meets the TESTING-RIGOR blocking bar.

## Scope: files affected

| Path | Action | Notes |
| --- | --- | --- |
| `scripts/validation/checks_eval.py` | create (A only) | New validator module: pattern match on branch diff, branch-diff evidence check, `SKIP_PROMPT_EVAL=1` bypass, CLI entry (`python3 scripts/validation/checks_eval.py --base-ref <ref>`) for the CI leg |
| `scripts/validation/pre_pr.py` | modify (A only) | Import `validate_prompt_eval_evidence` and add one `run_validation(...)` call in `main()` (wiring pattern at lines 272-276) |
| `tests/validation/test_checks_eval.py` | create (A only) | REQ-9 coverage; follow sibling `tests/validation/test_check_build_gates.py` conventions |
| `.github/workflows/ai-pr-quality-gate.yml` | modify (A only) | One thin step invoking the module (ADR-006); pin any new Action to SHA (none expected) |
| `.githooks/pre-commit` | modify (both) | Fix false enforcement comments at lines 2047-2049 and 2086; keep the advisory reminder block |
| `.agents/architecture/ADR-057-prompt-behavioral-evaluation.md` | modify (both) | Amendment section plus corrected Confirmation and Enforcement Path rows; fires adr-review |
| `.agents/sessions/<date>-session-<n>-*.json` | create (both) | Session log per ADR-007, created by `/session-init` |

Files NOT in scope here (owned by #3184): `.claude/hooks/PreToolUse/invoke_prompt_eval_gate.py`, its registrations in `.claude/settings.json:76-81` and `.claude/hooks/hooks.json:77-82`, the generated shim `src/copilot-cli/hooks/PreToolUse/invoke_prompt_eval_gate__Bash_gG_iI_tT_cC_oO_mM_mM_iI_tT_gG_iI_tT_cC_iI_g_ec29be.py`, `src/copilot-cli/hooks/PreToolUse/_manifest.json` lines 14 and 43, and the `tests/test_hook_plugin_guards.py:93` list entry. Because this issue touches nothing under `.claude/` or `src/copilot-cli/`, NO plugin manifest version bump is required (`.claude/rules/plugin-version-bump.md` binds only when those trees change) and NO shim regeneration (`python3 build/scripts/build_all.py`) is needed.

## Out of scope

- Deleting the inert hook, its two registrations, its Copilot shim, its `_manifest.json` entries, and its `tests/test_hook_plugin_guards.py` list entry: owned by #3184. Do not delete them here even though they are dead; double-deletion creates merge conflicts between the two PRs.
- Fixing the `{"decision":"deny"}` payload in place to make the hook block: rejected direction. The hook home costs a spawn on 100% of commits in two harnesses for a gate relevant to a minority of commits, and #3184 removes it.
- Any change to the eval runners themselves (`scripts/eval/eval-prompt-change.py`, `scripts/eval/eval-suite.py`), the acceptance gate math, or scenario storage. ADR-057 lines 78-130 govern those and they are not in question.
- Rehoming or changing other commit-matcher hooks (`invoke_security_commit_gate.py` is owned by #3189; `invoke_adr_review_guard.py` and `invoke_false_completion_gate.py` by #3196).
- CI automation of eval RUNS (invoking the LLM in CI). This issue gates on evidence artifacts only; running evals in CI remains the ADR-057 "Future" item and is a separate cost conversation.

## Implementation plan

Every command runs from the repo root `/` of a fresh clone of `rjmurillo/ai-agents`, default branch `main`. Direct commits to `main` are blocked; work on a feature branch.

1. Start a session and branch:

   ```bash
   git checkout -b feat/adr-057-eval-gate-rehome
   ```

   Run `/session-init` (Claude Code skill) to create the required session log under `.agents/sessions/` per ADR-007.

2. Decide the deliverable (REQ-1). Read the ADR and gather the data:

   ```bash
   cat .agents/architecture/ADR-057-prompt-behavioral-evaluation.md
   git log --oneline --since=2026-06-01 -- '.claude/skills/*/SKILL.md' '.claude/commands/' '.claude/agents/' | wc -l
   git log --oneline --since=2026-06-01 -- '.agents/eval-results/' | wc -l
   git log --oneline -10 -- .agents/eval-results/
   ```

   Decision rule: choose Deliverable A only if you conclude the regression-prevention driver (ADR-057 Decision Drivers 1-2, proven by Issue #1686's 20-point regression) still justifies blocking PRs, knowing current practice is 108 prompt-change commits to 1 evidence commit. If the evidence rules are obsolete in practice, choose Deliverable B: an ADR-057 amendment stating that automated enforcement is retired, evidence is produced on demand for high-risk prompt changes, and review is the human gate. State the choice and the data in the PR description. If B, skip steps 3-5 and go to step 6.

3. (A) Create `scripts/validation/checks_eval.py`:
   - `validate_prompt_eval_evidence(repo_root: Path, base_ref: str | None = None) -> bool`.
   - Compute changed files: `git diff --name-only <merge-base>...HEAD` using `_resolve_branch_base_ref` and `_run_subprocess` from `scripts/validation/checks_common.py` (do not shell out ad hoc; reuse the shared wrapper).
   - Match changed files against the three pattern lists copied character-for-character from the quotes above; docstring cites the source per `.claude/rules/canonical-source-mirror.md`.
   - Evidence: pass only if at least one changed file matches `^\.agents/eval-results/.*\.json$` (REQ-3). Honor `SKIP_PROMPT_EVAL=1` with a printed warning (REQ-5). On failure, print the matched files and the two runner commands (`python3 scripts/eval/eval-suite.py`, `python3 scripts/eval/eval-prompt-change.py --prompt <file> --scenarios <scenarios> --base-ref main`).
   - Add `if __name__ == "__main__":` CLI taking `--base-ref` so CI can invoke it as one command, exit codes per ADR-035 (0 ok, 1 logic, 2 config).

4. (A) Wire into `scripts/validation/pre_pr.py` `main()` next to the other fast checks (after the Python Syntax gate at lines 272-276): import from `checks_eval`, add one `run_validation("ADR-057 Prompt Eval Evidence", state, lambda: validate_prompt_eval_evidence(repo_root))` call. Write `tests/validation/test_checks_eval.py` per REQ-9.

5. (A) Add the CI leg to `.github/workflows/ai-pr-quality-gate.yml`: one step in the existing test job (near `python3 scripts/quality_gate/run_pytest.py`, line 235) running `python3 scripts/validation/checks_eval.py --base-ref "origin/${{ github.base_ref }}"`. No conditional logic in YAML beyond the step itself (ADR-006). Validate locally: `python3 scripts/validation/run_workflow_local_test.py` or `gh act` per `.claude/rules/ci-scripts.md`, and run the changed workflow before push per AGENTS.md.

6. (Both) Fix `.githooks/pre-commit` lines 2047-2049 and 2086: replace the "Claude Code hook ... provides the blocking enforcement" and "Claude hook provides enforcement" text with the true statement for your chosen deliverable.

7. (Both) Amend `.agents/architecture/ADR-057-prompt-behavioral-evaluation.md`: add a dated amendment section (follow the in-file precedent, "Acceptance Gate Relaxation (2026-06-01, Issue #2197)" at line 86), and correct the Confirmation row (line 222) and Enforcement Path (lines 243-247). This edit fires the adr-review skill; complete that review before PR.

8. Lint and validate:

   ```bash
   npx markdownlint-cli2 --fix ".agents/architecture/ADR-057-prompt-behavioral-evaluation.md"
   uv run pytest tests/validation/ -x
   python3 scripts/validation/pre_pr.py
   ```

9. Commit sequence (conventional commits, `Co-Authored-By:` trailer, 5-file cap per commit):
   - Commit 1 (A): `feat(validation): add ADR-057 prompt eval evidence check` covering `scripts/validation/checks_eval.py`, `tests/validation/test_checks_eval.py`.
   - Commit 2 (A): `feat(validation): wire ADR-057 eval gate into pre_pr and CI` covering `scripts/validation/pre_pr.py`, `.github/workflows/ai-pr-quality-gate.yml`.
   - Commit 3 (both): `docs(adr): record ADR-057 enforcement rehome and fix stale hook claims` covering `.agents/architecture/ADR-057-prompt-behavioral-evaluation.md`, `.githooks/pre-commit`, plus the session log.
   - For Deliverable B, only commit 3 ships (message: `docs(adr): retire ADR-057 automated eval enforcement`), with the amendment carrying the evidence.

10. Push and open a PR referencing this issue with `Fixes #<n>`, using the repo PR template.

## Acceptance criteria

- [ ] Deliverable decision recorded: PR description names A or B and cites the `git log` counts. Verify: read the PR description; both counts and a one-paragraph rationale are present.
- [ ] (A) Gate fails without evidence: on a scratch branch, `printf '\nscratch\n' >> .claude/commands/plan.md && git add .claude/commands/plan.md && git commit -m "test: scratch eval gate"` then `python3 scripts/validation/checks_eval.py --base-ref origin/main; echo $?` prints the matched file and `1`. Do not pass `--no-verify` (universal.md MUST NOT 2); the ADR-057 pre-commit reminder is non-blocking. Reset afterward with `git reset --hard HEAD~1`.
- [ ] (A) Gate passes with branch-diff evidence: same scratch branch plus `git add` of a new `.agents/eval-results/2026-07-17-test.json`, rerun prints `0`.
- [ ] (A) Bypass works loudly: `SKIP_PROMPT_EVAL=1 python3 scripts/validation/checks_eval.py --base-ref origin/main; echo $?` prints a warning containing `SKIP_PROMPT_EVAL` and exits `0`.
- [ ] (A) Wired into pre_pr: `python3 scripts/validation/pre_pr.py 2>&1 | grep "ADR-057"` shows the validation ran (PASS or FAIL line present).
- [ ] (A) CI leg present and thin: `grep -n "checks_eval" .github/workflows/ai-pr-quality-gate.yml` returns exactly one `run:` command line; no multi-line shell logic around it.
- [ ] (A) Tests pass: `uv run pytest tests/validation/test_checks_eval.py -x` exits 0 with positive, negative, and edge cases visible in `-v` output.
- [ ] (Both) False claims gone: `grep -n "Claude hook provides" .githooks/pre-commit` and `grep -n "invoke_prompt_eval_gate" .agents/architecture/ADR-057-prompt-behavioral-evaluation.md` return no lines asserting the hook blocks commits (historical references inside the amendment narrative are fine; present-tense enforcement claims are not).
- [ ] (Both) ADR amendment landed and adr-review completed: amendment section with 2026 date exists in ADR-057; the session log records the adr-review outcome.
- [ ] (Both) No em or en dashes introduced: `git diff origin/main | grep -P "[\x{2013}\x{2014}]"` returns nothing.
- [ ] Full gate green: `python3 scripts/validation/pre_pr.py` exits 0 on the branch.

## Verification

```bash
uv run pytest tests/validation/ -x
uv run ruff check scripts/validation/checks_eval.py tests/validation/test_checks_eval.py
python3 scripts/validation/checks_eval.py --base-ref origin/main; echo "exit=$?"
python3 scripts/validation/pre_pr.py
python3 scripts/validation/run_workflow_local_test.py
npx markdownlint-cli2 ".agents/architecture/ADR-057-prompt-behavioral-evaluation.md" ".githooks/pre-commit" 2>/dev/null || npx markdownlint-cli2 ".agents/architecture/ADR-057-prompt-behavioral-evaluation.md"
grep -n "Claude hook provides" .githooks/pre-commit || echo "stale claim removed: OK"
git diff origin/main | grep -P "[\x{2013}\x{2014}]" || echo "dash check: OK"
```

For Deliverable B, the `checks_eval` and workflow commands drop out; the remaining commands must all pass.

## Rollback

All changes are plain-text edits on a feature branch with no generated artifacts and no plugin manifests touched. Revert with `git revert <merge-commit-sha>` (or revert the individual commits in reverse order). Reverting restores: no eval check in `pre_pr.py`, no CI step, and the prior ADR/`.githooks/pre-commit` text. No shim regeneration (`python3 build/scripts/build_all.py`) is required because nothing under `.claude/` or `src/copilot-cli/` changes in this issue. If #3184 has NOT yet landed at rollback time, the repo returns exactly to today's state (inert hook plus advisory reminder); if it HAS landed, rollback of this issue leaves ADR-057 temporarily claiming an enforcement that no longer exists, so re-apply the ADR amendment commit (`git cherry-pick <commit-3-sha>`) or re-land quickly.

## Dependencies

- #3184: owns deletion of the inert hook, both registrations, the Copilot shim, `_manifest.json` entries, and the `tests/test_hook_plugin_guards.py:93` list entry. Soft ordering: either PR can land first because this issue quotes the hook's pattern lists verbatim above; if that issue lands first, retrieve the file with `git show <deletion-parent-sha>:.claude/hooks/PreToolUse/invoke_prompt_eval_gate.py` for reference. Landing delete-dead-hooks first is recommended so the false-claim fixes here (REQ-7, REQ-8) describe the end state.

## References

- `.agents/architecture/ADR-057-prompt-behavioral-evaluation.md`: the contract this issue enforces or amends (scope table lines 57-66, gate lines 78-111, Confirmation lines 216-247).
- `.claude/hooks/PreToolUse/invoke_security_commit_gate.py:10-18, 164-173`: in-repo documentation of the exit-0 deny-payload bug class (Issue #2521) that makes `invoke_prompt_eval_gate.py` inert.
- Issue #1686 (proved behavioral evals catch regressions structural tests miss; the +20-point case), Issue #1688 (methodology formalization), Issue #2197 (acceptance gate relaxation), Issue #2052 (pattern narrowing after agent move), Issue #2223 (pre_pr decomposition into `checks_*` modules), Issue #2521 (deny-payload bug).
- `.claude/rules/canonical-source-mirror.md`: verbatim-quote obligation for the pattern port.
- `.claude/rules/ci-scripts.md` and `.agents/architecture/ADR-006-thin-workflows-testable-modules.md`: thin-workflow constraint for the CI leg.
- `.agents/governance/TESTING-RIGOR.md`: blocking test bar for REQ-9.
- `.claude/rules/plugin-version-bump.md`: why no manifest bump is needed here.
- AGENTS.md: session gates, atomic commits, ADR edits fire adr-review.

Complexity: M

Part of #3197

### PR DESCRIPTION
**Pull request by rjmurillo**
## Summary

Corrects torn state in ADR-057 (Deliverable B of #3185): the ADR asserted a commit-time hook blocked prompt changes lacking behavioral-eval evidence. That hook was inert from the start (deny plus exit 0, ignored by the Claude Code harness) and was deleted in #3184, leaving dangling references and false "blocks commit" claims.

This PR amends ADR-057 to describe reality:

- Eval evidence is a MUST obligation on the contributor, but enforcement is advisory (PR review), not commit-blocking.
- One CI leg does block: the `/spec` behavioral eval runs on same-repo PRs that touch the spec command and fails the check on regression. It covers that one command only; every other prompt, skill, and agent file is advisory.
- Deliverable A (deterministic evidence enforcement) and running behavioral evals in CI are now distinguished as two separate future options, each with its own deferral rationale.

## Why Deliverable B, not A

#3185's title asks to build enforcement (Deliverable A). I recommend Deliverable B (advisory plus fix the torn refs) instead, because:

- The hook-ROI reduction program (#3197) is removing gates, not adding blocking hooks.
- The ADR's own Future path already defers CI eval automation pending a validated cost model.
- The acceptance gate inside the eval runner remains real and is used by the `/spec` CI leg and by contributors who run it before a PR.

Deliverable A stays documented as a future option if you prefer to build enforcement. This PR does not close #3185 by itself; it makes the decision explicit and removes the false claims. Close #3185 if you accept Deliverable B, or keep it open to pursue A.

## Review

Adversarial review by gpt-5.6-sol (non-Opus), 4 rounds, final verdict SHIP. It caught scope-precision issues (the `/spec` leg is a real blocking CI exception; Deliverable A is deterministic-only, not behavioral-eval-in-CI), which are corrected in the final revision.

## Verification

- Byte-verified 0 em-dash and 0 en-dash in the ADR body.
- ADR lives under a markdownlint-excluded location; no lint delta.

Refs #3185

## References

Paths named above that are not in this diff, for context only:

- see `scripts/validation/pre_pr.py`
- see `.github/workflows/slash-command-quality.yml`
- see `scripts/eval/eval-prompt-change.py`
- see `invoke_prompt_eval_gate.py` (deleted in #3184)

### CODE REVIEW

File: .agents/architecture/ADR-057-prompt-behavioral-evaluation.md [hunk 1]
```diff
@@ -217,9 +217,10 @@ Define targeted scenarios with expected verdicts. Run before/after comparison on
 
 ### Enforced (automated gates)
 
+These gates run inside `eval-prompt-change.py`. They fire whenever the eval runner is invoked: advisorily when a contributor runs it before a PR, and as a blocking check when the `/spec` CI leg (`slash-command-quality.yml`) invokes it on a same-repo PR that touches `spec.md`. No commit-time hook auto-invokes them (see Amendment 2026-07-22).
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .agents/architecture/ADR-057-prompt-behavioral-evaluation.md]: Fixed in 55fe329f. The ADR now describes the broader trigger: the leg fires when the PR changes the `/spec` command, its scenario set, or the eval runner (`spec_evals_changed`), while still evaluating the `/spec` command behavior only.

File: .agents/architecture/ADR-057-prompt-behavioral-evaluation.md [hunk 2]
```diff
@@ -78,7 +78,7 @@ Each evaluation consists of:
 The gate blocks regressions. It does not mandate an improvement on every edit. A prompt change passes behavioral evaluation when these criteria hold:
 
 1. `after_score >= before_score` (no regression on existing scenarios)
-2. No scenario flips from pass to fail. Every pass-to-fail flip is recorded in `regressions` and blocks the gate automatically; the automated gate has no mechanism to accept a "justified" regression. Landing a deliberate trade-off requires a human override (admin merge) with the rationale documented in the PR, not a gate bypass.
+2. No scenario flips from pass to fail. Every pass-to-fail flip is recorded in `regressions` and blocks the gate automatically; the gate has no mechanism to accept a "justified" regression. Where the gate runs as a blocking CI leg (currently the `/spec` eval in `slash-command-quality.yml`), landing a deliberate trade-off requires a human override (admin merge) with the rationale documented in the PR, not a gate bypass. For prompt files with no blocking CI leg, the gate is advisory and PR review carries the same judgment (see Amendment 2026-07-22).
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .agents/architecture/ADR-057-prompt-behavioral-evaluation.md]: Fixed in 78d8ac91. Criterion 2 now states the normal path: a deliberate behavior change lands by updating the scenario expectations alongside the prompt in the same change, so the gate passes without a bypass. Admin override is scoped to accepting a regression against unchanged expectations. Good catch, the prior wording was too absolute.

File: .agents/architecture/ADR-057-prompt-behavioral-evaluation.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .agents/architecture/ADR-057-prompt-behavioral-evaluation.md]: Correct, and thanks for the precise catch. The `/spec` blocking leg triggers more broadly than "touches `spec.md`". Per `slash-command-quality.yml` (the `spec_evals_changed` paths-filter on line 19), a same-repo PR fires the eval when it changes any of:

- `.claude/commands/spec.md`
- `tests/evals/spec-scenarios.json`
- `scripts/eval/eval-prompt-change.py`

In all three cases the leg evaluates the `spec.md` prompt behavior (the hardcoded `--prompt` argument) and fails the check on regression. The ADR amendment's substantive enforcement-scope claim (the leg covers `spec.md`'s behavior only) is accurate as written; this thread records the full trigger set for readers. Resolving.

File: .agents/architecture/ADR-057-prompt-behavioral-evaluation.md [hunk 3]
```diff
@@ -78,7 +78,7 @@ Each evaluation consists of:
...
 
 1. `after_score >= before_score` (no regression on existing scenarios)
-2. No scenario flips from pass to fail. Every pass-to-fail flip is recorded in `regressions` and blocks the gate automatically; the automated gate has no mechanism to accept a "justified" regression. Landing a deliberate trade-off requires a human override (admin merge) with the rationale documented in the PR, not a gate bypass.
+2. No scenario flips from pass to fail. Every pass-to-fail flip is recorded in `regressions` and blocks the gate automatically; the gate has no mechanism to accept a "justified" regression. Where the gate runs as a blocking CI leg (currently the `/spec` eval in `slash-command-quality.yml`), a deliberate behavior change normally lands by updating the scenario expectations alongside the prompt in the same change, so the new expectations move with the intended behavior and the gate passes without a bypass. Only accepting a regression against unchanged expectations requires a human override (admin merge) with the rationale documented in the PR. For prompt files with no blocking CI leg, the gate is advisory and PR review carries the same judgment (see Amendment 2026-07-22).
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .agents/architecture/ADR-057-prompt-behavioral-evaluation.md:81]: Valid consistency nit, and not merge-blocking. The reference is unambiguous either way: `slash-command-quality.yml` is the only workflow of that name in the repo, and the full path `.github/workflows/slash-command-quality.yml` already appears in this same ADR (Enforcement Path, line 247), so the file is easy to locate from any of the bare mentions on lines 81, 220, and 284.

Normalizing those three bare references to the full path is a cosmetic edit with no effect on any claim in the ADR, so I am deferring it rather than opening another amendment cycle on this docs-only PR. Resolving as acknowledged, not actioned.
